eng/code-quality-reports/src/main/resources/revapi/revapi.json

```diff
@@ -169,6 +169,30 @@
           "new": "method .* com\\.azure\\.resourcemanager\\..*",
           "justification": "resourcemanager interfaces are allowed to add methods."
         },
+        {
+          "ignore": true,
+          "code": "java.method.addedToInterface",
+          "new": "method <T> T com.azure.spring.data.cosmos.core.CosmosOperations::patch(T, com.azure.cosmos.models.CosmosPatchOperations, com.azure.cosmos.models.CosmosPatchItemRequestOptions)",
+          "justification": "Spring interfaces are allowed to add methods."
+        },
+        {
+          "ignore": true,
+          "code": "java.method.addedToInterface",
+          "new": "method <T> reactor.core.publisher.Mono<T> com.azure.spring.data.cosmos.core.ReactiveCosmosOperations::patch(T, com.azure.cosmos.models.CosmosPatchOperations, com.azure.cosmos.models.CosmosPatchItemRequestOptions)",
+          "justification": "Spring interfaces are allowed to add methods."
+        },
+        {
+          "ignore": true,
+          "code": "java.method.addedToInterface",
+          "new": "method <S extends T> S com.azure.spring.data.cosmos.repository.CosmosRepository<T, ID extends java.io.Serializable>::save(S, com.azure.cosmos.models.CosmosPatchOperations, com.azure.cosmos.models.CosmosPatchItemRequestOptions)",
+          "justification": "Spring interfaces are allowed to add methods."
+        },
+        {
+          "ignore": true,
```

Comment:
I don't get why this would be safe?

Reply:
In Spring nobody is implementing this interface except for us. The error is raised because it thinks by adding the new method, anybody who has implemented the interface will be broken; however with nobody implementing it in the case of Spring we can ignore it.

Reply:
Let's discuss in today's Java sync-up. I don't get why we make the interface public in the first place if we expect nobody to use it. Either public or it is not safe to add new members as you mentioned. Our API review reflects the same. If this is really a common spring pattern - is there a document describing this policy for Spring somewhere? Certainly possible - spring established its own release cycle etc. - so, not saying it doesn't exist. But I would like to take a look at it - also to be prepared if there any discussions with central folks about this.